### DISCUSSION TITLE
Drift files

### DISCUSSION THREAD

**Discussion by ghost** [simolus3/drift#1519] [Q&A] [answered]
I will be working on almost exclusively existing sqlite databases, I have a few questions about .drift files.

1. if I export the sql creation string and put it into a .drift file where can I specific the class name so like @DataClassName?

2. using the example of opening an existing database, I can't  see where the generated code goes or is none created?

3. can you use more than one .drift file e.g. one per table? if so how are they specified?

Thanks for a great piece of work.

**Comment by simolus3** [comment, marked as answer]
> if I export the sql creation string and put it into a .drift file where can I specific the class name so like

You can use `WITH` at the end of a table's definition to specify an existing row class that you wrote yourself. The Dart class used must be imported into the `drift` file for this.
You can also use `AS` at the end of a table to specify a custom data class name for drift to generate (e.g. `CREATE TABLE my_table (...) AS MyTable`). More docs on this are [here](https://drift.simonbinder.eu/docs/using-sql/moor_files/#existing-row-classes).

> using the example of opening an existing database, I can't see where the generated code goes or is none created?

None is generated for that example in particular. Opening databases is a runtime-only matter, so it's ignored by the generator. The example on existing databases assumes that you already have a drift database class with generated code.

> can you use more than one .drift file e.g. one per table? if so how are they specified?

Of course! Personally I like to split related queries into different files for instance.

The `include` parameter on `DriftDatabase` or `DriftAccessor` takes a set of drift files to import, so you can easily use more than one. If you `import` a drift file into another, that file will be implicitly added as well.
